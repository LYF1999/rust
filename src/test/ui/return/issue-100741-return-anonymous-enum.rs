struct Foo {}

fn foo<'a>() -> i32 | Vec<i32> | &str | &'a String | Foo {
    //~^ ERROR: anonymous enums are not supported
}

fn main() {}
