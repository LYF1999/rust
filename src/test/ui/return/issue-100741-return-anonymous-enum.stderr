error: anonymous enums are not supported
  --> $DIR/issue-100741-return-anonymous-enum.rs:3:17
   |
LL | fn foo<'a>() -> i32 | Vec<i32> | &str | &'a String | Foo {
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using enum as return type: `SomeEnum`
   |
   = note: enum SomeEnum<'lifetime,'a>{
               I32(i32)
               Vec(Vec<...>)
               StrRef(&'lifetime str)
               StringRef(&'a String)
               Foo(Foo)
           }
           

error: aborting due to previous error

